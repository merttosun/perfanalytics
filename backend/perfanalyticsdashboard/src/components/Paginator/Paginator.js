import React from "react";
import Pagination from "@material-ui/lab/Pagination";
import "./Paginator.css";
const Paginator = (props) => {
  const emitPageChange = (e, page) => {
    props.onPageChange(page);
  };
  return (
    <div className="paginator-wrapper">
      <Pagination
        className="paginator"
        count={props.pageCount}
        page={props.activePage}
        variant="outlined"
        shape="rounded"
        onChange={(e, page) => emitPageChange(e, page)}
      />
    </div>
  );
};

export default Paginator;
