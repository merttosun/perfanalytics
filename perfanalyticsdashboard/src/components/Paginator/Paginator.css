.paginator-wrapper {
  width: 100%;
  height: 50px;
  display: flex;
  justify-content: center;
  align-items: center;
}
.paginator {
  margin-top: 10px;
  max-width: 400px;
}
