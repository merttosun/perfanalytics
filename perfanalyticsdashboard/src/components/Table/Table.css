.resources-table {
  height: 500px;
  margin-top: 20px;
  border-radius: 6px;
  width: calc(40px + 90%);
  align-items: center;
  box-shadow: 0px 0px 11px 0px rgba(158, 152, 158, 1);
}
.header-row {
  background: #4285f4;
  color: white;
  display: flex;
  border-radius: 6px 6px 0 0;
  height: 30px;
}
th:first-child {
  flex-grow: 1;
  text-align: left;
  padding-left: 6px;
}
th {
  padding-top: 3px;
  width: 14%;
  text-align: left;
}
