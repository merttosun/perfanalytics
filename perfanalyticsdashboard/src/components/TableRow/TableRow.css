.table-row-wrapper,
.table-row {
  width: 100%;
  display: flex;
  border-bottom: 1px solid #c5c1c1;
  height: 40px;
  cursor: pointer;
}
.darker {
  background: rgb(226, 226, 233);
}
.row-column {
  width: 14%;
  display: flex;
  align-items: center;
}
.row-column:first-child {
  flex-grow: 1;
  justify-content: flex-start;
  padding-left: 10px;
}
.column-box {
  display: flex;
  flex-direction: column;
  width: 100%;
}
