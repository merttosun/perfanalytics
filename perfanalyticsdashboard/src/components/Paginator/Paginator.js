import React from "react";
import Pagination from "@material-ui/lab/Pagination";
import "./Paginator.css";
const Paginator = (props) => {
  return (
    <div className="paginator-wrapper">
      <Pagination
        className="paginator"
        count={props.pageCount}
        variant="outlined"
        shape="rounded"
      />
    </div>
  );
};

export default Paginator;
