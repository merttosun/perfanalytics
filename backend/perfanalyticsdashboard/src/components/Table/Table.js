import React, { useState } from "react";
import TableRow from "../TableRow/TableRow";
import Paginator from "../Paginator/Paginator";
import CircularProgress from "@material-ui/core/CircularProgress";
import "./Table.css";

const Table = (props) => {
  const [activePage, setActivePage] = useState(1);
  return (
    <div className="resources-table" data-testid="table">
      <tr className="header-row">
        <th>{props.headerColumnTxt_1}</th>
        <th>{props.headerColumnTxt_2}</th>
        <th>{props.headerColumnTxt_3}</th>
        <th>{props.headerColumnTxt_4}</th>
      </tr>
      <div className="table-content">
        {props.isLoading ? (
          <CircularProgress />
        ) : (
          props.data.map((i, index) => {
            return index >= (activePage - 1) * 10 && index < activePage * 10 ? (
              <TableRow
                className="table-row"
                key={index}
                even={index % 2 === 0}
                data={i}
              />
            ) : (
              ""
            );
          })
        )}
      </div>
      <Paginator
        pageCount={parseInt(props.data.length / 10)}
        activePage={activePage}
        disabled={props.isLoading}
        onPageChange={(page) => setActivePage(page)}
      />
    </div>
  );
};

export default Table;
