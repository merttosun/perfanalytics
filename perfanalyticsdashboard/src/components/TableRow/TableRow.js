import React from "react";
import "./TableRow.css";

const TableRow = (props) => {
  return (
    <tr className={props.even ? "table-row darker" : "table-row"}>
      <td className="row-column">qwe</td>
      <td className="row-column">asd</td>
      <td className="row-column">asd</td>
      <td className="row-column">asd</td>
    </tr>
  );
};

export default TableRow;
