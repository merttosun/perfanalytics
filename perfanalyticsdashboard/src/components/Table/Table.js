import React, { useState } from "react";
import TableRow from "../TableRow/TableRow";
import Paginator from "../Paginator/Paginator";
import CircularProgress from "@material-ui/core/CircularProgress";
import "./Table.css";

const Table = (props) => {
  const [activePage, setActivePage] = useState(1);
  return (
    <div className="resources-table">
      <tr className="header-row">
        <th>Url</th>
        <th>Duration</th>
        <th>Type</th>
        <th>Transfer Size</th>
      </tr>
      <div className="table-content">
        {props.isLoading ? (
          <CircularProgress />
        ) : (
          props.data.map((i, index) => {
            return index >= (activePage - 1) * 10 && index < activePage * 10 ? (
              <TableRow
                className="table-row"
                key={index}
                even={index % 2 === 0}
                data={i}
              />
            ) : (
              ""
            );
          })
        )}
      </div>
      <Paginator
        pageCount={props.data.length % 10}
        activePage={activePage}
        disabled={props.isLoading}
        onPageChange={(page) => setActivePage(page)}
      />
    </div>
  );
};

export default Table;
