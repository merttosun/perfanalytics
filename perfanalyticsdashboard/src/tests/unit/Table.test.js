import React from "react";
import ReactDom from "react-dom";
import { render } from "@testing-library/react";
import Table from "../../components/Table/Table";
import TableRow from "../TableRow/TableRow";
import Paginator from "../Paginator/Paginator";
import CircularProgress from "@material-ui/core/CircularProgress";
const tableBulkData = [
  {
    url: "http://localhost:3000/bundle.js",
    type: "script",
    duration: 27.704999985871837,
    transferSize: 946065,
  },
  {
    url: "http://localhost:3000/src/index.js",
    type: "other",
    duration: 4.944999993313104,
    transferSize: 2777,
  },
  {
    url: "http://localhost:3000/bundle.js",
    type: "script",
    duration: 27.065000002039596,
    transferSize: 946065,
  },
  {
    url: "http://localhost:3000/src/index.js",
    type: "other",
    duration: 5.3699999989476055,
    transferSize: 2777,
  },
  {
    url: "http://localhost:3000/bundle.js",
    type: "script",
    duration: 28.335000009974465,
    transferSize: 946065,
  },
];

it("Table renders without crashing", () => {
  const { getByTestId } = render(
    <Table isLoading={true} data={tableBulkData} />
  );
  const tableContainer = getByTestId("table");
  expect(tableContainer).toBeInTheDocument();
});

it("Table renders without crashing", () => {
  const { getByTestId } = render(
    <Table isLoading={true} data={tableBulkData} />
  );
  const tableContainer = getByTestId("table");
  expect(tableContainer).toBeInTheDocument();
});
