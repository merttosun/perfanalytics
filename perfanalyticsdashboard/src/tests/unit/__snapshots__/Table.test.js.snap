// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="resources-table"
  data-testid="table"
>
  <tr
    className="header-row"
  >
    <th>
      test url for snapshot
    </th>
    <th>
      test duration label for snapshot
    </th>
    <th>
      type
    </th>
    <th>
      Transfer Size2
    </th>
  </tr>
  <div
    className="table-content"
  >
    <div
      className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
      role="progressbar"
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </div>
  </div>
  <div
    className="paginator-wrapper"
  >
    <nav
      aria-label="pagination navigation"
      className="MuiPagination-root paginator"
    >
      <ul
        className="MuiPagination-ul"
      >
        <li>
          <button
            aria-label="Go to previous page"
            className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined MuiPaginationItem-rounded Mui-disabled Mui-disabled"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            type="button"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiPaginationItem-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
              />
            </svg>
          </button>
        </li>
        <li>
          <button
            aria-label="Go to next page"
            className="MuiButtonBase-root MuiPaginationItem-root MuiPaginationItem-page MuiPaginationItem-outlined MuiPaginationItem-rounded Mui-disabled Mui-disabled"
            disabled={true}
            onBlur={[Function]}
            onClick={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={-1}
            type="button"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root MuiPaginationItem-icon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
              />
            </svg>
          </button>
        </li>
      </ul>
    </nav>
  </div>
</div>
`;
